File: cashtab/src/wallet/context.js
// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import React from 'react';
import PropTypes from 'prop-types';
import useWallet from 'wallet/useWallet';

export const WalletContext = React.createContext();

export const WalletProvider = ({ chronik, agora, ecc, children }) => {
    const wallet = useWallet(chronik, agora, ecc);
    return (
        <WalletContext.Provider value={wallet}>
            {children}
        </WalletContext.Provider>
    );
};

WalletProvider.propTypes = {
    chronik: PropTypes.object,
    agora: PropTypes.object,
    ecc: PropTypes.object,
    children: PropTypes.node,
};

File: src/node/context.cpp
// Copyright (c) 2019 The Bitcoin Core developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

#include <node/context.h>

#include <addrman.h>
#include <banman.h>
#include <interfaces/chain.h>
#include <net.h>
#include <net_processing.h>
#include <node/kernel_notifications.h>
#include <scheduler.h>
#include <txmempool.h>
#include <validation.h>

namespace node {
NodeContext::NodeContext() {}
NodeContext::~NodeContext() {}
} // namespace node
